radicle-node/src/wire/new.rs

```diff
@@ -0,0 +1,273 @@
+                self.inner.disconnected(
+                    peer_addr.id(),
+                    &DisconnectReason::Protocol(service::DisconnectReason::Error(
+                        session::Error::Misbehavior,
+                    )),
+                );
+            }
+            SessionEvent::ConnectionFailure(err) => {
+                let peer_addr = addr.expect_peer_addr();
+                self.inner.disconnected(
+                    peer_addr.id(),
+                    &DisconnectReason::ConnectionError(Arc::new(err)),
+                );
+            }
+            SessionEvent::Disconnected => {
+                let peer_addr = addr.expect_peer_addr();
+                self.inner.disconnected(
+                    peer_addr.id(),
+                    &DisconnectReason::Protocol(service::DisconnectReason::Peer),
+                );
+            }
+        }
+    }
+
+    fn handle_command(&mut self, cmd: Self::Command) {
+        self.inner.command(cmd)
+    }
+
+    fn handle_error(
+        &mut self,
+        err: reactor::Error<net::SocketAddr, XkAddr<NodeId, net::SocketAddr>>,
+    ) {
+        match err {
+            reactor::Error::ListenerUnknown(id) => {
+                log::error!("asking for unknown listener {}", id);
+            }
+            reactor::Error::PeerUnknown(id) => {
+                log::error!("asking for unknown peer {}", id);
+            }
+            reactor::Error::PeerDisconnected(addr, err) => {
+                log::error!("the peer {} got disconnected with {}", addr, err);
+                self.inner_queue
+                    .push_back(reactor::Action::UnregisterTransport(addr))
+            }
+        }
+    }
+}
+
+impl<R, S, W, G> Iterator for Wire<R, S, W, G>
+where
+    R: routing::Store,
+    S: address::Store,
+    W: WriteStorage + 'static,
+    G: Signer,
+{
+    type Item = reactor::Action<NetAccept<Session>, NetTransport<Session, Framer>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(event) = self.inner_queue.pop_front() {
+            return Some(event);
+        }
+
+        match self.inner.next() {
+            Some(Io::Write(id, msgs)) => {
+                log::debug!("Sending {} messages to {}", msgs.len(), id);
+                Some(reactor::Action::Send(id, msgs))
+            }
+            Some(Io::Event(e)) => {
+                log::warn!(
+                    "Received an event, while events are not handled in the current version"
+                );
+                // TODO: Handle the events
+                None
+            }
+            Some(Io::Connect(id, addr)) => {
+                let NetAddr { host, port } = *addr;
+                let socket_addr = match host {
+                    HostAddr::Ip(ip) => net::SocketAddr::new(ip, addr.port()),
+                    HostAddr::Dns(_) => todo!(),
+                    _ => self.proxy,
+                };
+
+                self.inner.attempted(&socket_addr);
```

Comment:
We shouldn't call this here, it should only be called if we are able to create the socket. I guess it's replaced by `connecting` anyway, which we didn't have before?